### DISCUSSION TITLE
how can i convert DynamicImage to an base64 format

### DISCUSSION THREAD

**Discussion by aisuta** [image-rs/image#1814] [Q&A]
i want to tranfer an image through tcp
thank you very much

**Comment by fintelia** [comment]
To send an image in base64, you probably first want to encode it in a specific file format like PNG:
```rust
let mut bytes: Vec<u8> = Vec::new();
img.write_to(&mut Cursor::new(&mut bytes), image::ImageOutputFormat::Png)?;
```
Then it is just a matter of base64 encoding the `Vec<u8>` (which you can probably find another crate for)
